### ISSUE TITLE
List of languages pending colour correction or addition

### ISSUE THREAD

**Issue by Alhadis** [github/linguist#4506]
#4417 was closed yesterday, so I felt it might be beneficial to keep a record of languages whose colours are incorrect (according to their branding or community's "preferred" colour). That way they won't fade into obscurity after being blocked for so long.

Now, these fixes are blocked because of an ugly tangle of colour-proximity violations, which was previously discussed in [`#4291`](https://github.com/github/linguist/issues/4291#issuecomment-452278428). Since it's unclear when (or even *if*) a decision will be made on GitHub's part, I'm compiling them here so nobody forgets. :+1: The list is incomplete and I've probably missed a lot, so feel free to make amendments as you see fit.

* [ ] **Asymptote:** `#ff0000` (source: [logo](http://asymptote.sourceforge.net/logo.svg))
* [ ] **Apex:** `#1797c0` (source: [`#4417`](https://github.com/github/linguist/issues/4417))
* [ ] **Blade:** `#f7523f`
* [ ] **CSON:** `#244776` (source: "inherited" from CoffeeScript)
* [ ] **Eiffel:** `#4d6977` (source: [logo](https://upload.wikimedia.org/wikipedia/commons/6/6a/Eiffel_logo.svg))
* [ ] **Grammatical Framework**: `#ff0000` (source: [logo](http://www.grammaticalframework.org/))
* [ ] **Haml:** `#ece2a9`
* [ ] **Handlebars:** `#f7931e`
* [ ] **Latte:** `#f2a542`
* [ ] **Less:** `#1d365d`
* [ ] **Marko:** `#42bff2`
* [ ] **Markdown:** `#083fa1` (source: [original colour value](https://github.com/github/linguist/pull/3112#issuecomment-232171926))
* [ ]  **mIRC Script Language (mSL)**: `#3D57C3` (source: "m" in [logo](https://www.mirc.com/images/indexlogorb.gif))
* [ ] **PostScript:** `#da291c` (source: [`#3141`](https://github.com/github/linguist/pull/3141))
* [ ] **Prisma**: `#0C344B`
* [ ] **Pug:** `#a86454`
* [ ] **SCSS:** `#c6538c`
* [ ] **Sass:** `#a53b70`
* [ ] **Slim:** `#2b2b2b`
* [ ] **SourcePawn:** `#f69e1d` (source: [logo](https://www.sourcemod.net/))
* [ ] **Stylus:** `#ff6347`
* [ ] **SVG:** `#ff9900` (source: [logo](https://www.w3.org/Graphics/SVG/svglogo.svg))
* [ ] **Twig:** `#c1d026`
* [ ] **V:** `#4f87c4` (source: [logo](https://vlang.io/img/favicon.png))
* [ ] **Vue:** `#2c3e50` (source: [`#4448`](https://github.com/github/linguist/issues/4448#issuecomment-470070769))
* [ ] **YAML:** `#cb171e` (source: [logo](https://github.com/yaml))

## Sidenotes

-	Some of these are `data` or `prose` languages, which normally *don't* display
	colours, but some are of relevance now that the `linguist-detectable` attribute
	can reveal them in a project's statistics-bar. I'm being conservative and only
	adding colours for data formats (like SVG or YAML) that have an obvious colour
	defined by their logos.

-	Many of these are markup languages waiting to be degrouped. Degrouping them now
	with missing/incorrect colours would be disruptive to many users.

-	I'm reluctant to tweak the colours of "well-known/famous" languages like JavaScript
	or TypeScript, which have the highest degree of visibility on GitHub. Even if they
	currently deviate only a few shades from their official colours, I predict astute
	users with oddly-calibrated monitors will notice the shift immediately.

**Comment by johnjcamilleri** [comment]
I would like to add:

- [ ] **Grammatical Framework**: `#ff0000` (source: [logo](http://www.grammaticalframework.org/))

**Comment by Alhadis** [comment]
Added, thank you!

**Comment by X6B0ISI09wD4** [comment]
[ ] **SourcePawn:** `#f69e1d` (source: [logo](https://www.sourcemod.net/))

**Comment by david-tamar** [comment]
**Eiffel**: `#4d6977` (source: [logo](https://upload.wikimedia.org/wikipedia/commons/6/6a/Eiffel_logo.svg))

The current color for the **Eiffel** programming language is `#946d57`, what's the source of that color?

The current color of **Ada** (`#02f88c`) is also quite puzzling to me, what's the origin of it? Is it from the cover of the [1981 Ada book by I. C. Pyle](https://images-na.ssl-images-amazon.com/images/I/41xK8YpRH3L._SX363_BO1,204,203,200_.jpg)?

**Comment by Alhadis** [comment]
@david-tamar Regarding the unknown colour sources: it's quite likely they were selected at random, possibly because the official colour was unknown or unavailable.

Added Eiffel to the OP.

**Comment by Alhadis** [comment]
@lildude Any update?

**Comment by lildude** [comment]
Nope. Still waiting on GitHub Design to give me the 👍 or 👎 on my request to drop the proximity enforcement.

**Comment by Alhadis** [comment]
Hopefully they can prioritise this or something, because there's a *lot* of stuff blocked on this... 😞

**Comment by Teniclub** [comment]
> [# 4417](https://github.com/github/linguist/pull/4417)ถูกปิดเมื่อวานนี้ดังนั้นฉันรู้สึกว่ามันอาจเป็นประโยชน์ในการเก็บบันทึกภาษาที่มีสีไม่ถูกต้อง (อ้างอิงจากแบรนด์ของพวกเขาหรือสีที่ "ต้องการ" ของชุมชน) ด้วยวิธีนี้พวกเขาจะไม่จางหายไปสู่ความสับสนหลังจากถูกบล็อกมานาน
> 
> [`#4291`](https://github.com/github/linguist/issues/4291#issuecomment-452278428)ตอนนี้แก้ปัญหาเหล่านี้จะถูกบล็อกเพราะยุ่งเหยิงน่าเกลียดของการละเมิดสีใกล้ชิดซึ่งได้รับการกล่าวถึงก่อนหน้านี้ใน เพราะมันไม่ชัดเจนเมื่อ (หรือแม้กระทั่ง_ถ้า_ ) ตัดสินใจที่จะทำในส่วน GitHub ของฉันรวบรวมพวกเขาที่นี่จึงไม่มีใครลืม รายการไม่สมบูรณ์และฉันอาจจะพลาดมากดังนั้นอย่าลังเลที่จะทำการแก้ไขตามที่คุณเห็นสมควร
> 
> * [ ]  **เส้นกำกับ:**  (ที่มา: [โลโก้](http://asymptote.sourceforge.net/logo.svg) )`#ff0000`[](http://asymptote.sourceforge.net/logo.svg)
> * [ ]  **เอเพ็กซ์:**  (ที่มา: )`#1797c0`[`#4417`](https://github.com/github/linguist/issues/4417)
> * [ ]  **ใบมีด:** `#f7523f`
> * [ ]  **CSON:**  (แหล่งที่มา: "สืบทอด" จาก CoffeeScript)`#244776`
> * [ ]  **ไอเฟล:**  (แหล่งที่มา: [โลโก้](https://upload.wikimedia.org/wikipedia/commons/6/6a/Eiffel_logo.svg) )`#4d6977`[](https://upload.wikimedia.org/wikipedia/commons/6/6a/Eiffel_logo.svg)
> * [ ]  **กรอบไวยากรณ์** : (ที่มา: [โลโก้](http://www.grammaticalframework.org/) )`#ff0000`[](http://www.grammaticalframework.org/)
> * [ ]  **Haml:** `#ece2a9`
> * [ ]  **แฮนด์:** `#f7931e`
> * [ ]  **ลาเต้:** `#f2a542`
> * [ ]  **หัก:** `#1d365d`
> * [ ]  **Marko:** `#42bff2`
> * [ ]  **Markdown:**  (แหล่งที่มา: [ค่าสีดั้งเดิม](https://github.com/github/linguist/pull/3112#issuecomment-232171926) )`#083fa1`[](https://github.com/github/linguist/pull/3112#issuecomment-232171926)
> * [ ]  **PostScript:**  (ที่มา: )`#da291c`[`#3141`](https://github.com/github/linguist/pull/3141)
> * [ ]  **ปั๊ก:** `#a86454`
> * [ ]  **SCSS:** `#c6538c`
> * [ ]  **Sass:** `#a53b70`
> * [ ]  **บาง:** `#2b2b2b`
> * [ ]  **SourcePawn:**  (แหล่งที่มา: [โลโก้](https://www.sourcemod.net/) )`#f69e1d`[](https://www.sourcemod.net/)
> * [ ]  **สไตลัส:** `#ff6347`
> * [ ]  **SVG:**  (แหล่งที่มา: [โลโก้](https://www.w3.org/Graphics/SVG/svglogo.svg) )`#ff9900`[](https://www.w3.org/Graphics/SVG/svglogo.svg)
> * [ ]  **ทวิก:** `#c1d026`
> * [ ]  **V:**  (แหล่งที่มา: [โลโก้](https://vlang.io/img/favicon.png) )`#4f87c4`[](https://vlang.io/img/favicon.png)
> * [ ]  **Vue:**  (ที่มา: )`#2c3e50`[`#4448`](https://github.com/github/linguist/issues/4448#issuecomment-470070769)
> * [ ]  **YAML:**  (แหล่งที่มา: [โลโก้](https://github.com/yaml) )`#cb171e`[](https://github.com/yaml)
> 
> ## Sidenotes
> * สิ่งเหล่านี้บางส่วนเป็น`data`หรือ`prose`ภาษาซึ่งโดยปกติ_จะไม่_แสดง
>   สี แต่บางอันก็มีความเกี่ยวข้องกันในขณะนี้ซึ่ง`linguist-detectable`แอตทริบิวต์
>   สามารถเปิดเผยได้ในแถบสถิติของโครงการ ฉันเป็นคนอนุรักษ์นิยมและ
>   เพิ่มเฉพาะสีสำหรับรูปแบบข้อมูล (เช่น SVG หรือ YAML) ที่มีสีที่ชัดเจนซึ่ง
>   กำหนดโดยโลโก้ของพวกเขา
> * สิ่งเหล่านี้จำนวนมากเป็นมาร์กอัป langugages ที่รอการจัดกลุ่ม การจัดกลุ่ม
>   สีที่หายไป / ไม่ถูกต้องในตอนนี้จะทำให้ผู้ใช้หลายคนหยุดชะงัก
> * ฉันลังเลที่จะปรับแต่งสีของภาษา "ที่มีชื่อเสียง / โด่งดัง" เช่น JavaScript 
>   หรือ TypeScript ซึ่งมีการแสดงผลระดับสูงที่สุดใน GitHub แม้ว่าพวกเขาจะ
>   เบี่ยงเบนเพียงไม่กี่เฉดสีจากสีทางการของพวกเขาฉันคาดการณ์
>   ผู้ใช้ที่ชาญฉลาดด้วยจอภาพที่ปรับเทียบอย่างผิดปกติจะสังเกตเห็นการเปลี่ยนแปลงในทันที

https://github.com/github/linguist/issues/4506#issue-436984670

**Comment by david-tamar** [comment]
> Nope. Still waiting on GitHub Design to give me the +1 or -1 on my request to drop the proximity enforcement.

I'd like to thumbs-up your issue, can you give me a link?

**Comment by Alhadis** [comment]
I think it's internal. As in, visible only to GitHub staff.

If it were public, I'm sure we'd be seeing links to this issue being referenced in this thread's activity feed. E.g., stuff like *"@lildude referenced this issue on 19 Sep"* and what-have-you.

**Comment by Weakky** [comment]
I'd like to add Prisma to the list 🙏

- [x] **Prisma**: `#0C344B` (colliding with DataWeave `#003a52`)

Thanks!

**Comment by Alhadis** [comment]
Hey @Weakky, 👋

I'll add it once you've actioned https://github.com/github/linguist/pull/4668#discussion_r332994275. 😉

**Comment by david-tamar** [comment]
I wish to suggest a new color `#57068C` for Ada based on the historical birth place of it's principal free implementation, that is [New York University's **Ada Translator**](https://en.wikipedia.org/wiki/GNAT). The color is taken directly from New York University's official color, logo and seal.

Wikipedia cites that the US Department of Defense (which sponsored Ada in the first place) awarded an exclusive contract to NYU to implement a free implementation of Ada strictly under FSF's GPL.

NYU's **Ada Translator** (widely abbreviated today as **GNAT**, where the G stands for GNU, by association with FSF's GNU GPL) is in fact the only standard free implementation of Ada that is both officially sponsored by the US government, and freely available for everyone.

It's therefore most likely that every Ada project on GitHub uses NYU's implementation, since it's officially distributed on most Linux distributions like Debian, Fedora, Ubuntu and Red Hat. It's also the go-to implementation for Windows and Mac users.

AdaCore, a major contributor of NYU's Ada, is also targeting enterprise and corporate clients with a full featured infrastructure and IDE that ship with NYU's GNAT as the default implementation.

I searched the internet for historical logos or mascots of Ada, but I did not find any official logos or mascots that featured any other colors than plain black on white, and dark blue, and yet most of those commercial graphics (either from AdaCore or communities) came way after NYU's work on Ada. It's also worthwhile to note that apparently the current color for Ada was chosen arbitrarily.

**Comment by Alhadis** [comment]
@david-tamar This is worth being filed as a new issue, especially because [there are guidelines in place](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#changing-the-color-associated-with-a-language) for changing the colours of well-known languages.

This issue is only for colours that are unambiguously correct/official, but are blocked from use entirely due to another colour being slightly too similar.

**Comment by david-tamar** [comment]
@Alhadis, It turns out the current color is actually not arbitrary as I thought. A user on Reddit's Ada community has shown me the connection between Ada and the green color. Ada was actually code-named “_Green_” by its original author in its very early stage at least 12 years before New York University worked on the Ada Translator, so never mind.

> Ichbiah's team submitted a language design labelled "Green" to a competition to choose the United States Department of Defense's embedded programming language. When Green was selected in 1978, he continued as chief designer of the language, now named "Ada".

**Comment by Alhadis** [comment]
So, I'm not sure when this happened exactly, but it looks like GitHub is now displaying language colours underneath code-search results as well:

<img src="https://user-images.githubusercontent.com/2346707/67149227-66abfc00-f2f4-11e9-8d2c-891d1ceeca43.png" alt="Figure 1" />

Which is going to leave a lot of grey for `data`/`prose` languages...

**Comment by gen-angry** [comment]
A bit later after it's approval but I'd like to add:

* [ ]  **mIRC Script Language (mSL)**: `#3D57C3` (source: [logo](https://www.mirc.com/images/indexlogorb.gif), going off the M)

It is currently: `#926059` because it was the first one I could find free.

Thanks :)

**Comment by Alhadis** [comment]
Hey @gen-angry, sorry for the slow response. 👋 I've added it to the list, thanks!

**Comment by ObserverOfTime** [comment]
Could you also add:

* [ ] **Svelte**: `#ff3e00` (source: [logo](https://github.com/sveltejs/branding/blob/master/svelte-logo.svg))

It's currently grouped under HTML.

**Comment by Alhadis** [comment]
@ObserverOfTime If it's grouped under HTML, the colour won't be visible anyway (because such "child" languages always inherit their parent's colour, irrespective of `.gitattributes` files).

**Comment by ObserverOfTime** [comment]
I am aware but I think it should be ungrouped. It's similar to Vue which isn't part of the HTML group. Should I have mentioned it in #4291?

**Comment by Alhadis** [comment]
> Should I have mentioned it in #4291?

Yes, actually. 👍

**Comment by lildude** [comment]
From https://github.com/github/linguist/issues/4969

> Or better yet, we don't wait and just go ahead and crush it. It'd serve GitHub's design-team right for taking (literally) years to decide if similar-looking colours from two-facelifts ago are still a problem. 😜

I'm thiiiiiiis close to doing it and dealing with the fallout if and when it happens 😬. If I were to 🔥 it, do we want to allow duplicate colours? I'm thinking not at first as we'll end up with a million identical reds or blues.

**Comment by Alhadis** [comment]
> If I were to 🔥 it, do we want to allow duplicate colours? I'm thinking not at first as we'll end up with a million identical reds or blues

I'll copy+paste my [earlier response](https://github.com/github/linguist/issues/4969#issuecomment-68340390) so folks here will see it:

> Yes, of course. Otherwise, we'd end up with squabbles over which language is more deserving of `#f00`, and we'd have no argument to justify this outside of a popularity contest. If a repository has identical-looking shades of red and blue… congratulations, that's what colour-blind users are already seeing. 😉

**Comment by lildude** [comment]
🤦 Of course. I didn't think about that.

**Comment by Alhadis** [comment]
On second thought, let's just get it over with. #4291 will take all of 10 minutes to address, and the changes will be noticed—and warmly-welcomed—by virtually every web developer on GitHub.  Since we're on the verge of a new release, I say: **_LET'S DO THIS SHIT._**

**Comment by Alhadis** [comment]
Right. Done. See #4978, #4979, #4980.
